### DISCUSSION TITLE
Fuzzy fish-like filename completion

### DISCUSSION THREAD

**Discussion by gagbo** [marlonrichert/zsh-autocomplete#503] [Q&A] [answered]
Hello,

I'm converting from fish to zsh, and I'm trying to get my configuration to replicate the fuzzy filename completion after the last `/`. I.e. I'd like the first completion in the screenshot to be `config-xfk.el`, and not `config-denote.el` (it may look similar to #421 but not really, I still have the issue even with that zstyle set)

![image](https://user-images.githubusercontent.com/10496163/198870171-2d3436fb-84ef-456a-b70e-453cc07a84b0.png)

Do you know how I'm supposed to change zstyle to fix that? I suppose that I should be using [the fuzzy completer style](https://github.com/marlonrichert/zsh-autocomplete/blob/f07efda31f27446c8940b069d054bb8b0f61e026/scripts/.autocomplete.config#L66), but I didn't really find which zstyle incantation to use to target file completions, and to tell it to use that fuzzy (or loose) style.

I don't understand how the common substring is chosen, I thought it would just be `"config"`

FTR, my .zshrc:

```zsh
# Set the zsh home and data path variables
DATA_HOME="${XDG_DATA_HOME:-$HOME/.local/share}"
##################################

# Ensure zsh-snap

## Download Znap, if it's not there yet.
[[ -f "${DATA_HOME}/zsh-snap/znap.zsh" ]] ||
    git clone --depth 1 -- \
        https://github.com/marlonrichert/zsh-snap.git "${DATA_HOME}/zsh-snap"

source "${DATA_HOME}/zsh-snap/znap.zsh"  # Start zsh-snap

## Znap plugins
znap source romkatv/powerlevel10k

### zsh-autocomplete config
zstyle ':autocomplete:*' widget-style menu-complete
zstyle ':completion:*:paths' path-completion yes

znap source marlonrichert/zsh-autocomplete
znap source zsh-users/zsh-autosuggestions
znap source zsh-users/zsh-syntax-highlighting
znap install zsh-users/zsh-completions
znap install Emiller88/guix-zsh-completions

# To customize prompt, run `p10k configure` or edit ${ZDOTDIR}/.p10k.zsh.
[[ ! -f "${ZDOTDIR}/.p10k.zsh" ]] || source "${ZDOTDIR}/.p10k.zsh"
```
(I removed my path hacks and powerlevel10k config because I don't think they are relevant, but the full config is [on Sourcehut](https://git.sr.ht/~gagbo/dotfiles/tree/master/item/.config/zsh/.zshrc))

Thanks for this plugin and your time,
Gerry

**Comment by marlonrichert** [comment, marked as answer]
Try adding the following _after_ sourcing Autocomplete:
```zsh
zstyle ':completion:*' completer \
    _expand _complete _complete:-loose _complete:-fuzzy _ignored
zstyle ':autocomplete:*' min-input 1
```

**Reply by gagbo** [reply]
That works! Thanks, I assume that I needed to tell completion to use the "loose" style before fuzzy ?

**Reply by marlonrichert** [reply]
No, I just removed the `_approximate` completer, which tries to do spelling correction. I’m planning to move that to a separate key binding in future anyway.
